doc/manual/source/man/dnst-keyset.rst

```diff
@@ -0,0 +1,279 @@
+First the entire zone is transfer from the primary nameserver listed in the
+SOA record.
+Then all relevant signatures are checked if they have the expected key tags.
+The maximum TTL in the zone is recorded to be reported.
+Finally, all addresses of listed nameservers are checked to see if they
+have a SOA serial that is greater or equal to the one that was checked.
+
+Automation of ``cache-expired1`` and ``cache-expired2`` is enabled by the
+``expire`` boolean.
+When enabled, the cron subcommand simply checks if enough time has passed
+to invoke ``cache-expired1`` or ``cache-expired2``.
+
+Finally the ``done`` boolean enabled automation of the ``roll-done`` step.
+This automation is very similar to the ``report`` automation.
+This only difference is that the Wait actions are automated so propagation
+is track but no TTL is reported.
+
+Fine grained control of over automation makes it possible to automate
+KSK or algorithm except that they are start manually.
+Or let a key roll progress automatically except that the ``cache-expired``
+steps are manual to be able insert extra manual steps.
+
+The ``report`` and ``done`` automations require that keyset has network access
+to all nameservers of the zone and all nameservers of the parent.
+
+HSM Support (KMIP)
+~~~~~~~~~~~~~~~~~~
+
+Importing Keys
+~~~~~~~~~~~~~~
+
+There are three basic ways to import exiting keys: public-key,
+a public/private key pair from files or a public/private key pair in an HSM.
+
+A public key can only be import from a file.
+When the key is imported the name of the file is stored in the key set and
+the key will be included in DNSKEY RRset.
+This is useful for certain migration and to manually implement a 
+multi-signer DNSSEC signing setup.
+Note that automation does not work for the case.
+
+A public/private key pair can be imported from files.
+It sufficient to give the name of the file that holds the public key if
+the filename ends in ``.key`` and the filename is the private key is the
+same except that it ends in ``.private``. 
+If this is not the case then the private key filename can be specified 
+separately.
+
+Importing a public/private key on an HSM require specifying the KMIP
+server ID, the ID of the public key, the ID of the private key, the
+DNSSEC algorithm of the key and the flags (typically 256 for a ZSK and
+257 for a KSK).
+
+Normally, keyset assumes ownership of any keys it hold.
+This mean that when a key is deleted from the key set, the keyset subcommand
+will also delete the files that hold the public and private or delete the
+key from the HSM.
+
+For an import public/private key pair this is considered too dangerous
+because another signer may need the keys.
+For this reason keys are imported in so-called ``decoupled`` state.
+When a decoupled key is deleted, only the reference to the key is deleted
+from the key set, the underlying keys are left untouched.
+There is a ``--coupled`` option to tell keyset to take ownership of the key.
```

Comment:
```suggestion
For an imported public/private key pair this is considered too dangerous
because another signer may need the keys.
For this reason keys are imported in so-called ``decoupled`` state.
When a decoupled key is deleted, only the reference to the key is deleted
from the key set, the underlying keys are left untouched.
There is a ``--coupled`` option to tell keyset to take ownership of the key.
```